### PR TITLE
Fixes #128 Handle form input validation

### LINKED ISSUE #128: Handle and validate form input both in the front-end and back-end
### This is a
- [x] Bug report
- [ ] Feature request

Following up on https://github.com/fossasia/yaydoc/pull/94#issuecomment-307365666

### Current behaviour

Form inputs are not validated and handled
### Expected behaviour

Proper form validation both in the front-end and the back-end

### PR DESCRIPTION
**Pull request by imujjwal96**
## Related Issue

Issue: #128 
## Screenshots (if appropriate):
asd

## Types of changes

- [x] Bug fix 
- [ ] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

File: backend/generator.js [hunk 1]
```diff
@@ -3,6 +3,10 @@ var exports = module.exports = {};
 var uuidV4 = require("uuid/v4");
 
 exports.executeScript = function (socket, formData) {
+  if (!isValid(formData)) {
```
Comment (pri22296) [inline backend/generator.js]: If we are laready validating the input in form.js, is it required here again?

Comment (imujjwal96) [inline backend/generator.js]: A user can always manipulate the validation on the front end,

For example, 
image

Comment (pri22296) [inline backend/generator.js]: I agree btw manipulating the validation would only lead to a build error and nothing else. There shouldn't be any security concerns. But if you still want validation on the server's end, could you not duplicate the isValid method?

Comment (imujjwal96) [inline backend/generator.js]: What do you mean? Duplicating as in calling that function?

Comment (pri22296) [inline backend/generator.js]: isValid is defined both in generator.js and form.js

Comment (imujjwal96) [inline backend/generator.js]: there is a difference between them, the backend script doesn't return a message as in the front end. Handling flash messages from the backend seemed strenous

Comment (pri22296) [inline backend/generator.js]: I can see that but still the function almost duplicated. That's not a good idea. Maybe return a error code from the function and use it to display message in the front end. That way you'll only need to define the function once.

Comment (imujjwal96) [inline backend/generator.js]: i dont think if it is possible to call a backend function on the frontend

Comment (imujjwal96) [inline backend/generator.js]: I have made the changes. Thanks for the inspiration :D

**Comment by mariobehling** [comment]
This project is starting to take shape! Very good!

Please resolve merge conflicts.

**Comment by imujjwal96** [comment]
Appreciated @mariobehling 
Anyway, resolved the merge conflicts

**Pull request merged**